ledger/eval.go

```diff
@@ -1043,10 +1025,15 @@ func eval(ctx context.Context, l ledgerForEvaluator, blk bookkeeping.Block, vali
 
 	// If validating, do final block checks that depend on our new state
 	if validate {
-		close(txvalidator.txgroups)
-		err, gotErr := <-txvalidator.done
-		if gotErr && err != nil {
-			return StateDelta{}, err
+		// wait for the validation to complete.
+		select {
```

Comment:
don't need a one option select{}, just do the receive statement?

Reply:
hmm.. no, it's a bug:
it should also test for `ctx.Done()` since the context might be canceled. Will be fixed shortly.